src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -992,49 +1008,128 @@ public static void onWorldLoad(@NotNull final World world)
         }
     }
 
-    public static boolean backupColonyData()
+    /**
+     * Initiates a timer to schedule the colony backup task
+     * backupDelay of 0 disables automatic backup
+     */
+    public static void scheduleColonyBackup()
     {
-        if (numWorldsLoaded > 0 && saveNeeded)
+        final int backupDelay = Configurations.administration.colonyBackupTimeMins;
+        if (backupDelay > 0)
         {
-            saveColonies(false);
+            backupTimerTask = new backupColonyTimerTask();
+            backupTimer.schedule(backupTimerTask, backupDelay * 60 * 1000);
         }
+    }
+
+    /**
+     * Task to be called from backupTimer, triggers main backup method
+     * by overriding run() method of base TimerTask
+     */
 
-        try(FileOutputStream fos = new FileOutputStream(getBackupSaveLocation(new Date())))
+    private static class backupColonyTimerTask extends TimerTask
+    {
+        @Override
+        public void run()
         {
-            @NotNull final File saveDir = new File(DimensionManager.getWorld(0).getSaveHandler().getWorldDirectory(), FILENAME_MINECOLONIES_PATH);
-            final ZipOutputStream zos = new ZipOutputStream(fos);
+            Log.getLogger().info(String.format("Initiating scheduled backup"));
+            backupColonyData();
+        }
+    }
 
-            for (int i = 1; i < colonies.getTopID() + 1; i++)
+    /**
+     * Thread safe method to lock backups so only one backup may run at a time
+     *
+     * @return boolean indicating whether safe to initiate backup
+     */
+    synchronized private static boolean isSafeToBackup()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename this method to start with "is" or "has". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2047)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AModifiersOrderCheck)